core/src/main/java/com/adobe/aem/commons/assetshare/content/impl/datasources/AssetRenditionsDataSource.java

```diff
@@ -76,32 +79,47 @@ protected final void doGet(SlingHttpServletRequest request, SlingHttpServletResp
                 PN_EXCLUDE_ASSETRENDITIONS);
 
         for (final AssetRenditionDispatcher assetRenditionDispatcher : assetRenditions.getAssetRenditionDispatchers()) {
-            if (assetRenditionDispatcher.isHidden()) {
-                log.debug("Skip adding AssetRenditionDispatcher factory [ {} ] to Data Source as it has been marked as hidden via configuration", assetRenditionDispatcher.getName());
-                continue;
-            } else if (excludeAssetRenditionDispatchers.contains(assetRenditionDispatcher.getName())) {
-                log.debug("Skip adding AssetRenditionDispatcher factory [ {} ] to Data Source as it has been excluded via configuration", assetRenditionDispatcher.getName());
-                continue;
-            }
 
-            assetRenditionDispatcher.getOptions().entrySet().stream()
-                    .filter(entry -> !excludeAssetRenditions.contains(entry.getValue()))
-                    .filter(entry -> !data.containsValue(entry.getValue()))
-                    .forEach(entry -> {
-                        String label = entry.getKey();
-                        String value = entry.getValue();
+            if (acceptsAssetRenditionDispatcher(allowedAssetRenditionTypes,
+                    excludeAssetRenditionDispatchers,
+                    assetRenditionDispatcher)) {
+
+                assetRenditionDispatcher.getOptions().entrySet().stream()
+                        .filter(entry -> !excludeAssetRenditions.contains(entry.getValue()))
+                        .filter(entry -> !data.containsValue(entry.getValue()))
+                        .forEach(entry -> {
+                            String label = entry.getKey();
+                            String value = entry.getValue();
 
-                        if (cfg.add_assetrenditiondispatcher_to_label()) {
-                            label += " (" + assetRenditionDispatcher.getLabel() + ")";
-                        }
+                            if (cfg.add_assetrenditiondispatcher_to_label()) {
+                                label += " (" + assetRenditionDispatcher.getLabel() + ")";
+                            }
 
-                        data.put(label, value);
-                    });
+                            data.put(label, value);
+                        });
+            }
         }
 
         dataSourceBuilder.build(request, data);
     }
 
+    private boolean acceptsAssetRenditionDispatcher(List<String> allowedAssetRenditionTypes, Set<String> excludeAssetRenditionDispatchers, AssetRenditionDispatcher assetRenditionDispatcher) {
+        if (!assetRenditionDispatcher.getTypes().isEmpty() &&
+                !allowedAssetRenditionTypes.isEmpty() &&
+                Collections.disjoint(assetRenditionDispatcher.getTypes(), allowedAssetRenditionTypes)) {
```

Comment:
Can this be rewritten to have less `!` logic? Isn't the equivalent:

`assetRenditionDispatcher.getTypes().isEmpty() ||
                allowedAssetRenditionTypes.isEmpty() ||
                !Collections.disjoint(assetRenditionDispatcher.getTypes(), allowedAssetRenditionTypes)`


it's just hard to follow what's going on here.

Reply:
You'd have to NOT that entire condition; which to me is even harder to grok. 
I'll add a comment explaining this condition to help mitigate confusion.